src/app/courses/courses.component.ts

```diff
@@ -172,6 +177,36 @@ export class CoursesComponent implements OnInit, OnChanges, AfterViewInit, OnDes
     });
   }
 
+  @HostListener('window:resize', ['$event'])
+  onResize(event) {
+    this.updateDeviceType();
+  }
+
+  updateDeviceType() {
+    const width = window.innerWidth;
+
+    if (width < 768) {
+      this.isMobile = true;
+      this.isTablet = false;
+      this.titleLength = 25; // Limit for mobile devices
+    } else if (width >= 768 && width < 1024) {
+      this.isMobile = false;
+      this.isTablet = true;
+      this.titleLength = 40; // Limit for tablets
+    } else {
+      this.isMobile = false;
+      this.isTablet = false;
+      this.titleLength = 100; // Limit for desktop
+    }
+  }
```

Comment:
I think there's a global device size, I think it would be nice to use it, in case we change something for the device type in terms of resolution, we don't have to hunt down hardcoded values inside

you can find it on _variables.scss, 
// Our screen sizes
$screen-md: 1000px;
$screen-sm: 780px;